packages/grafana-data/src/vector/FunctionalVector.ts

```diff
@@ -0,0 +1,73 @@
+import { vectorToArray } from './vectorToArray';
+import { Vector } from '../types';
+
+export abstract class FunctionalVector<T = any> implements Vector<T>, Iterable<T> {
+  abstract get length(): number;
+
+  abstract get(index: number): T;
+
+  // Implement "iterator protocol"
+  *iterator() {
+    for (let i = 0; i < this.length; i++) {
+      yield this.get(i);
+    }
+  }
+
+  // Implement "iterable protocol"
+  [Symbol.iterator]() {
+    return this.iterator();
+  }
+
+  forEach(iterator: (row: T) => void) {
+    for (let i = 0; i < this.length; i++) {
```

Comment:
should be able to just iterate here:  for(const item of this)

Reply:
you got to let me keep for loops somewhere!